File: versioned_docs/version-1.14.x/examples/py/src/data_management_delete_data.py

from reduct import Client, Bucket


async def main():
    # Create a client with the base URL and API token
    async with Client("http://localhost:8383", api_token="my-token") as client:
        # Get bucket to remove
        bucket: Bucket = await client.get_bucket("bucket-to-remove")

        # Delete only entry with name "example-entry"
        await bucket.remove_entry("example-entry")

        # Remove entire bucket
        await bucket.remove()


if __name__ == "__main__":
    import asyncio

    asyncio.run(main())

File: versioned_docs/version-1.18.x/examples/cpp/src/data_management_delete_data.cc
#include <reduct/client.h>
#include <cassert>
#include <chrono>
#include <iostream>
#include <thread>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

int main() {
    // Create a client with the server URL
    auto client = IClient::Build("http://127.0.0.1:8383", {
            .api_token = "my-token"
    });

    // Get bucket to remove
    auto [bucket, get_err] = client->GetBucket("bucket-to-remove");
    assert(get_err == Error::kOk);

    // Delete only entry with name "example-entry"
    auto remove_entry_err = bucket->RemoveEntry("example-entry");
    assert(remove_entry_err == Error::kOk);

    // Wait for the storage to finish removing blocks
    std::this_thread::sleep_for(std::chrono::seconds(1));

    // Remove entire bucket
    auto remove_err = bucket->Remove();
    assert(remove_err == Error::kOk);
}
